spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,77 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phase, such as dev, testing, live, etc. 
+
+=== Specify Config Profile
+
+The config profile can be specified via the property `mp.config.profile`, which can be set in any of the configuration sources. The value of the property can contain only characters that are valid for config property names. 
+This is because the name of the profile is directly stored in the name of the config property. It can be set when starting your application. e.g.
+
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+The value of the property `mp.config.profile` shouldn't be updated after the application is started. It's only read once during the application start up. If the property value is modified afterwards, the behavior is undefined and any changes to its value made later can be ignored by the implementation. 
+
+The value of the property `mp.config.profile` specifies a single active profile. Only one profile can be active at a time. Commas in the value have no special meaning. For example, if the value of `mp.config.profile` is `testing,live`,  a single profile named `testing,live` is active instead of two active profiles. 
+If the property `mp.config.profile` is specified in multiple config sources, the value of the property is determined following the same rules as other configuration properties, which means the value in the config source with the highest ordinal wins.
+
+=== How Config Profile works
+
+The configuration property that utilizes the Config Profile is called a "profile-specific" property. A "profile-specific" property name consists of the following sequence: `% <profile name>.<original property name>`.
+
+Conforming implementations are required to search for a configuration source with the highest ordinal (priority) that provides either the property name or the "profile-specific" property name. 
+If the configuration source provides the "profile-specific" name, the value of the "profile-specific" property will be returned. If it doesn't contain the "profile-specific" name, the value of the plain property will be returned. 
+
+
+For an instance, a config source can be specified as follows.
+
+[source, text]
+----
+%dev.vehicle.name=car
+%live.vehicle.name=train
+%testing.vehicle.name=bike
+vehicle.name=lorry
+----
+
+A config property associated with the Config Profile can be looked up as shown below.
+
+[source, text]
+----
+@Inject @ConfigProperty(name="vehicle.name") String vehicleName;
+----
+
+[source, text]
+----
+String vehicleName = ConfigProvider.getConfig().getValue("vehicle.name", String.class);
+----
+
+If the property `mp.config.profile` is set to `dev`, the property `%dev.vehicle.name` is the Active Property. An active property overrides the properties in the same config source. 
```

Comment:
This is exactly the other way around as any other established solutions do it. Everyone else is using postfixing, e.g. vehicle.name.dev. The reason is the following: 
* you can easily group configuration that way
* you can have multiple lookup paths e.g. a dev.oracle and dev.postgres postfix.

Reply:
I think prefix or postfix both works. SmallRye Config went with prefix. I think what you listed above can be done using prefix as well.
`%dev.vehicle.name` stands out more than `vehicle.name.dev` though.

Reply:
The issue with postfixing is that since a profile name can be anything you can encounter clashes. So `vehicle.name.something`, how do I know if `something` refers to a profile and I should set up `vehicle.name`, or it is an actual property I should setup `vehicle.name.something.something`?